### ISSUE TITLE
v1.56 - build error

### ISSUE THREAD

**Issue by zdm** [rurban/Filter#3]
```
Entering Filter-1.56
Checking configure dependencies from META.json
Checking if you have ExtUtils::MakeMaker 6.58 ... Yes (7.24)
Running Makefile.PL
Configuring Filter-1.56 ... Checking if your kit is complete...
Looks good
Warning: LINKTYPE set to '', no longer necessary
Writing MYMETA.yml and MYMETA.json
Writing MYMETA.yml and MYMETA.json
Writing MYMETA.yml and MYMETA.json
Writing MYMETA.yml and MYMETA.json
Generating a Unix-style Makefile
Writing Makefile for Filter::Util::Call
Writing MYMETA.yml and MYMETA.json
Checking dependencies from MYMETA.json ...
Checking if you have ExtUtils::MakeMaker 0 ... Yes (7.24)
OK
Building and testing Filter-1.56 ... cp lib/Filter/cpp.pm blib/lib/Filter/cpp.pm
cp perlfilter.pod blib/lib/Filter/Util/perlfilter.pod
cp lib/Filter/exec.pm blib/lib/Filter/exec.pm
cp lib/Filter/sh.pm blib/lib/Filter/sh.pm
make[1]: Entering directory `/root/.cpanm/work/1485030324.7/Filter-1.56/Call'
cp Call.pm ../blib/lib/Filter/Util/Call.pm
Running Mkbootstrap for Call ()
chmod 644 "Call.bs"
"/usr/local/perlbrew/perls/perl-5.24.1/bin/perl" -MExtUtils::Command::MM -e 'cp_nonempty' -- Call.bs ../blib/arch/auto/Filter/Util/Call/Call.bs 644
"/usr/local/perlbrew/perls/perl-5.24.1/bin/perl" "/usr/local/perlbrew/perls/perl-5.24.1/lib/5.24.1/ExtUtils/xsubpp"  -typemap '/usr/local/perlbrew/perls/perl-5.24.1/lib/5.24.1/ExtUtils/typemap' -typemap '/root/.cpanm/work/1485030324.7/Filter-1.56/Call/typemap'  Call.xs > Call.xsc
mv Call.xsc Call.c
cc -c   -fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -D_FORTIFY_SOURCE=2 -O2   -DVERSION=\"1.56\" -DXS_VERSION=\"1.56\" -fPIC "-I/usr/local/perlbrew/perls/perl-5.24.1/lib/5.24.1/x86_64-linux-ld/CORE"  -D_NOT_CORE Call.c
rm -f ../blib/arch/auto/Filter/Util/Call/Call.so
cc  -shared -O2 -L/usr/local/lib -fstack-protector-strong Call.o  -o ../blib/arch/auto/Filter/Util/Call/Call.so  \
      \

chmod 755 ../blib/arch/auto/Filter/Util/Call/Call.so
make[1]: Leaving directory `/root/.cpanm/work/1485030324.7/Filter-1.56/Call'
make[1]: Entering directory `/root/.cpanm/work/1485030324.7/Filter-1.56/Exec'
cp Exec.pm ../blib/lib/Filter/Util/Exec.pm
Running Mkbootstrap for Exec ()
chmod 644 "Exec.bs"
"/usr/local/perlbrew/perls/perl-5.24.1/bin/perl" -MExtUtils::Command::MM -e 'cp_nonempty' -- Exec.bs ../blib/arch/auto/Filter/Util/Exec/Exec.bs 644
"/usr/local/perlbrew/perls/perl-5.24.1/bin/perl" "/usr/local/perlbrew/perls/perl-5.24.1/lib/5.24.1/ExtUtils/xsubpp"  -typemap '/usr/local/perlbrew/perls/perl-5.24.1/lib/5.24.1/ExtUtils/typemap'  Exec.xs > Exec.xsc
mv Exec.xsc Exec.c
cc -c   -fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -D_FORTIFY_SOURCE=2 -O2   -DVERSION=\"1.55\" -DXS_VERSION=\"1.55\" -fPIC "-I/usr/local/perlbrew/perls/perl-5.24.1/lib/5.24.1/x86_64-linux-ld/CORE"   Exec.c
rm -f ../blib/arch/auto/Filter/Util/Exec/Exec.so
cc  -shared -O2 -L/usr/local/lib -fstack-protector-strong Exec.o  -o ../blib/arch/auto/Filter/Util/Exec/Exec.so  \
      \

chmod 755 ../blib/arch/auto/Filter/Util/Exec/Exec.so
make[1]: Leaving directory `/root/.cpanm/work/1485030324.7/Filter-1.56/Exec'
make[1]: Entering directory `/root/.cpanm/work/1485030324.7/Filter-1.56/decrypt'
cp decrypt.pm ../blib/lib/Filter/decrypt.pm
Running Mkbootstrap for decrypt ()
chmod 644 "decrypt.bs"
"/usr/local/perlbrew/perls/perl-5.24.1/bin/perl" -MExtUtils::Command::MM -e 'cp_nonempty' -- decrypt.bs ../blib/arch/auto/Filter/decrypt/decrypt.bs 644
"/usr/local/perlbrew/perls/perl-5.24.1/bin/perl" "/usr/local/perlbrew/perls/perl-5.24.1/lib/5.24.1/ExtUtils/xsubpp"  -typemap '/usr/local/perlbrew/perls/perl-5.24.1/lib/5.24.1/ExtUtils/typemap'  decrypt.xs > decrypt.xsc
mv decrypt.xsc decrypt.c
cc -c   -fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -D_FORTIFY_SOURCE=2 -O2   -DVERSION=\"1.55\" -DXS_VERSION=\"1.55\" -fPIC "-I/usr/local/perlbrew/perls/perl-5.24.1/lib/5.24.1/x86_64-linux-ld/CORE"  -DBYPASS decrypt.c
rm -f ../blib/arch/auto/Filter/decrypt/decrypt.so
cc  -shared -O2 -L/usr/local/lib -fstack-protector-strong decrypt.o  -o ../blib/arch/auto/Filter/decrypt/decrypt.so  \
      \

chmod 755 ../blib/arch/auto/Filter/decrypt/decrypt.so
make[1]: Leaving directory `/root/.cpanm/work/1485030324.7/Filter-1.56/decrypt'
make[1]: Entering directory `/root/.cpanm/work/1485030324.7/Filter-1.56/tee'
cp tee.pm ../blib/lib/Filter/tee.pm
Running Mkbootstrap for tee ()
chmod 644 "tee.bs"
"/usr/local/perlbrew/perls/perl-5.24.1/bin/perl" -MExtUtils::Command::MM -e 'cp_nonempty' -- tee.bs ../blib/arch/auto/Filter/tee/tee.bs 644
"/usr/local/perlbrew/perls/perl-5.24.1/bin/perl" "/usr/local/perlbrew/perls/perl-5.24.1/lib/5.24.1/ExtUtils/xsubpp"  -typemap '/usr/local/perlbrew/perls/perl-5.24.1/lib/5.24.1/ExtUtils/typemap'  tee.xs > tee.xsc
mv tee.xsc tee.c
cc -c   -fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -D_FORTIFY_SOURCE=2 -O2   -DVERSION=\"1.55\" -DXS_VERSION=\"1.55\" -fPIC "-I/usr/local/perlbrew/perls/perl-5.24.1/lib/5.24.1/x86_64-linux-ld/CORE"   tee.c
rm -f ../blib/arch/auto/Filter/tee/tee.so
cc  -shared -O2 -L/usr/local/lib -fstack-protector-strong tee.o  -o ../blib/arch/auto/Filter/tee/tee.so  \
      \

chmod 755 ../blib/arch/auto/Filter/tee/tee.so
make[1]: Leaving directory `/root/.cpanm/work/1485030324.7/Filter-1.56/tee'
make[1]: Entering directory `/root/.cpanm/work/1485030324.7/Filter-1.56/Call'
"/usr/local/perlbrew/perls/perl-5.24.1/bin/perl" -MExtUtils::Command::MM -e 'cp_nonempty' -- Call.bs ../blib/arch/auto/Filter/Util/Call/Call.bs 644
make[1]: Leaving directory `/root/.cpanm/work/1485030324.7/Filter-1.56/Call'
make[1]: Entering directory `/root/.cpanm/work/1485030324.7/Filter-1.56/Exec'
"/usr/local/perlbrew/perls/perl-5.24.1/bin/perl" -MExtUtils::Command::MM -e 'cp_nonempty' -- Exec.bs ../blib/arch/auto/Filter/Util/Exec/Exec.bs 644
make[1]: Leaving directory `/root/.cpanm/work/1485030324.7/Filter-1.56/Exec'
make[1]: Entering directory `/root/.cpanm/work/1485030324.7/Filter-1.56/decrypt'
"/usr/local/perlbrew/perls/perl-5.24.1/bin/perl" -MExtUtils::Command::MM -e 'cp_nonempty' -- decrypt.bs ../blib/arch/auto/Filter/decrypt/decrypt.bs 644
make[1]: Leaving directory `/root/.cpanm/work/1485030324.7/Filter-1.56/decrypt'
make[1]: Entering directory `/root/.cpanm/work/1485030324.7/Filter-1.56/tee'
"/usr/local/perlbrew/perls/perl-5.24.1/bin/perl" -MExtUtils::Command::MM -e 'cp_nonempty' -- tee.bs ../blib/arch/auto/Filter/tee/tee.bs 644
make[1]: Leaving directory `/root/.cpanm/work/1485030324.7/Filter-1.56/tee'
make[1]: Entering directory `/root/.cpanm/work/1485030324.7/Filter-1.56/Call'
"/usr/local/perlbrew/perls/perl-5.24.1/bin/perl" -MExtUtils::Command::MM -e 'cp_nonempty' -- Call.bs ../blib/arch/auto/Filter/Util/Call/Call.bs 644
No tests defined for Filter::Util::Call extension.
make[1]: Leaving directory `/root/.cpanm/work/1485030324.7/Filter-1.56/Call'
make[1]: Entering directory `/root/.cpanm/work/1485030324.7/Filter-1.56/Exec'
"/usr/local/perlbrew/perls/perl-5.24.1/bin/perl" -MExtUtils::Command::MM -e 'cp_nonempty' -- Exec.bs ../blib/arch/auto/Filter/Util/Exec/Exec.bs 644
No tests defined for Filter::Util::Exec extension.
make[1]: Leaving directory `/root/.cpanm/work/1485030324.7/Filter-1.56/Exec'
make[1]: Entering directory `/root/.cpanm/work/1485030324.7/Filter-1.56/decrypt'
"/usr/local/perlbrew/perls/perl-5.24.1/bin/perl" -MExtUtils::Command::MM -e 'cp_nonempty' -- decrypt.bs ../blib/arch/auto/Filter/decrypt/decrypt.bs 644
No tests defined for Filter::decrypt extension.
make[1]: Leaving directory `/root/.cpanm/work/1485030324.7/Filter-1.56/decrypt'
make[1]: Entering directory `/root/.cpanm/work/1485030324.7/Filter-1.56/tee'
"/usr/local/perlbrew/perls/perl-5.24.1/bin/perl" -MExtUtils::Command::MM -e 'cp_nonempty' -- tee.bs ../blib/arch/auto/Filter/tee/tee.bs 644
No tests defined for Filter::tee extension.
make[1]: Leaving directory `/root/.cpanm/work/1485030324.7/Filter-1.56/tee'
PERL_DL_NONLAZY=1 "/usr/local/perlbrew/perls/perl-5.24.1/bin/perl" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')" t/*.t
t/call.t .................... ok
t/cpp.t ..................... ok
Wide character in print at decrypt382.tst line 6.
t/decrypt.t .................
Failed 1/7 subtests
t/exec.t .................... ok
t/order.t ................... ok
t/rt_101033.t ............... ok
t/rt_54452-rebless.t ........ ok
t/sh.t ...................... ok
t/tee.t ..................... ok
t/z_kwalitee.t .............. skipped: This test is only run for the module author
t/z_manifest.t .............. skipped: requires a git checkout and a unix for git and diff
t/z_meta.t .................. skipped: This test is only run for the module author
t/z_perl_minimum_version.t .. skipped: Author tests not required for installation
t/z_pod-coverage.t .......... skipped: This test is only run for the module author
t/z_pod.t ................... ok
Failed 1/15 test programs. 1/65 subtests failed.

Test Summary Report
-------------------
t/decrypt.t               (Wstat: 0 Tests: 7 Failed: 1)
  Failed test:  7
Files=15, Tests=65,  1 wallclock secs ( 0.05 usr  0.04 sys +  0.81 cusr  0.28 csys =  1.18 CPU)
Result: FAIL
make: *** [test_dynamic] Error 255
! Installing R/RU/RURBAN/Filter-1.56.tar.gz failed. See /root/.cpanm/work/1485030324.7/build.log for details. Retry with --force to force install it.
FAIL
```

**Comment by rurban** [comment]
No, that's a test error with a new UTF-8 locale sensitive test.
Unfortunately not repro for me.
http://www.cpantesters.org/distro/F/Filter.html?oncpan=1&distmat=1&version=1.55&grade=3#

Working on fixes, see subsequent updates.

**Comment by zdm** [comment]
Also, this issue is present with strawberry perl 5.24.0.

**Comment by zdm** [comment]
And, when installing package on centos under docker.

**Comment by rurban** [comment]
For the record: fixed with 1.57

**Comment by zdm** [comment]
Yes, I already tested, thank you.
